staging/src/k8s.io/apimachinery/pkg/runtime/error.go

```diff
@@ -120,3 +120,43 @@ func IsMissingVersion(err error) bool {
 	_, ok := err.(*missingVersionErr)
 	return ok
 }
+
+// strictDecoderError is a base error type that is returned by a strict Decoder such
+// as UniversalStrictDecoder.
+type strictDecoderError struct {
+	message      string
+	gvk          schema.GroupVersionKind
+	originalData []byte
+}
+
+// NewStrictDecoderError creates a new strictDecoderError object
+func NewStrictDecoderError(message string, gvk schema.GroupVersionKind, originalData []byte) error {
+	return &strictDecoderError{
+		message:      message,
+		gvk:          gvk,
+		originalData: originalData,
+	}
+}
+
+func (e *strictDecoderError) Error() string {
+	return fmt.Sprintf("strict decoder error for %#v: %s", e.gvk, e.message)
+}
+
+// GVK returns the GVK that was extracted when Decoding using a strict Decoder.
+func (e *strictDecoderError) GVK() schema.GroupVersionKind {
+	return e.gvk
+}
+
+// OriginalData returns the original byte slice input that was passed to a strict Decoder.
+func (e *strictDecoderError) OriginalData() []byte {
```

Comment:
where is this used? Why do we need it?

Reply:
I don't know if we strictly need it. If you want we can drop it for now. It's one of those things that _can_  be useful

Reply:
Like `missingKindErr` error in this file. But i would like move `originalData` to the sub-Error struct, `UnknownFieldError` and `DuplicateFieldError`(those all remove now) , may be add in the future

Reply:
It does not make much sense to have these two method on a strict decoder error, but return an untyped error when there is a (non-strict) decoding error. This is not uniform.